app/features/vods/VodRepository.server.ts

```diff
@@ -1,5 +1,258 @@
+import { sql } from "kysely";
 import { db } from "~/db/sql";
+import type { Tables } from "~/db/tables";
+import type {
+	MainWeaponId,
+	ModeShort,
+	StageId,
+} from "~/modules/in-game-lists/types";
+import { weaponIdToArrayWithAlts } from "~/modules/in-game-lists/weapon-ids";
+import {
+	dateToDatabaseTimestamp,
+	dayMonthYearToDatabaseTimestamp,
+} from "~/utils/dates";
+import invariant from "~/utils/invariant";
+import { selectPlayers } from "~/utils/kysely.server";
+import { VODS_PAGE_BATCH_SIZE } from "./vods-constants";
+import type { ListVod, VideoBeingAdded, Vod } from "./vods-types";
+import {
+	extractYoutubeIdFromVideoUrl,
+	hoursMinutesSecondsStringToSeconds,
+} from "./vods-utils";
 
 export function deleteById(id: number) {
 	return db.deleteFrom("UnvalidatedVideo").where("id", "=", id).execute();
 }
+
+export async function findVods({
+	weapon,
+	mode,
+	stageId,
+	type,
+	userId,
+	limit = VODS_PAGE_BATCH_SIZE,
+}: {
+	weapon?: MainWeaponId;
+	mode?: ModeShort;
+	stageId?: StageId;
+	type?: Tables["Video"]["type"];
+	userId?: number;
+	limit?: number;
+}): Promise<Array<ListVod>> {
```

Comment:
It might be better to spin off findVodsByUserId as its on function from here. I know the existing code was not like this but I think with the better code reusability it might read better. Or what do you think?